pages/guides/developing/encrypted-vod.en-US.mdx

```diff
@@ -0,0 +1,186 @@
+---
+title: 'Upload encrypted files to Livepeer'
+description: 'Learn how to upload and playback an encrypted video asset'
+---
+
+import { Callout } from 'nextra-theme-docs';
+
+# Encrypted VOD with Livepeer
+
+Livepeer enables you to upload encrypted video content or import it from decentralized storages. It creates a playback URL that can be covered by playback policies and access control.
+
+<Callout>
+	This guide utilize snippets in Golang as examples.
+</Callout>
+
+## Step 1: Generate an encryption key
+
+First, we generate a 256-bit encryption key to encrypt the file. 
+
+```go filename="main.go"
+// Generate random 256-bit encryption key
+key := make([]byte, 32)
+_, err = rand.Read(key)
+if err != nil {
+  log.Fatalf("Error generating random key: %v", err)
+}
+```
+
+## Step 2: Encrypt your video using the encryption key
+
+To encrypt a video file with the key we just generated, we follow these steps:
+
+1. Generate a random 128-bit initialization vector (IV).
+2. Pad the data using PKCS#7 padding.
+3. Encrypt the data using AES-CBC.
+
+```go filename="main.go"
+// Read the input file
+data, err := ioutil.ReadFile("your/local/path/source.mp4")
+if err != nil {
+  log.Fatalf("Error reading input file: %v", err)
+}
+
+// Generate random 128-bit IV
+iv := make([]byte, aes.BlockSize)
+_, err = rand.Read(iv)
+if err != nil {
+  log.Fatalf("Error generating random IV: %v", err)
+}
+
+// Pad the data using PKCS#7 padding
+paddedData := pkcs7Pad(data, aes.BlockSize)
+
+// Encrypt the data using AES-CBC with PKCS#7 padding
+block, err := aes.NewCipher(key)
+if err != nil {
+  log.Fatalf("Error creating AES cipher: %v", err)
+}
+ciphertext := make([]byte, len(paddedData))
+mode := cipher.NewCBCEncrypter(block, iv)
+mode.CryptBlocks(ciphertext, paddedData)
+
+// Write the encrypted data and IV to the output file
+encryptedData := make([]byte, aes.BlockSize+len(ciphertext))
+copy(encryptedData[:aes.BlockSize], iv)
+copy(encryptedData[aes.BlockSize:], ciphertext)
+err = ioutil.WriteFile("your/local/path/encrypted.mp4", encryptedData, 0644)
+if err != nil {
+  log.Fatalf("Error writing encrypted data to output file: %v", err)
+}
+```
+<Callout>
+  Currently Livepeer supports only video content encrypted using AES-CBC with PKCS#7 padding.
+</Callout>
```

Comment:
Since our focus is this being done from the browser, this doc can be simplified to refer to WebCrypto browser lib instead. WDYT?
```suggestion
<Callout>
  Currently Livepeer supports only video content encrypted using [SubtleCrypto.encrypt](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/encrypt) with `AES-CBC` algorithm, which is also the encryption used by other web3 protocols like [Lit](https://developer.litprotocol.com/sdk/explanation/encryption/). It can implemented in other environments with regular AES-CBC encryption using PKCS#7 padding.
</Callout>
```